src/circuit/gadget/utilities/cond_swap.rs

```diff
@@ -179,18 +179,18 @@ impl<F: FieldExt> CondSwapChip<F> {
             let one = Expression::Constant(F::one());
 
             // a_swapped - b ⋅ swap - a ⋅ (1-swap) = 0
-            // This checks that `a_swapped` is equal to `y` when `swap` is set,
+            // This checks that `a_swapped` is equal to `b` when `swap` is set,
             // but remains as `a` when `swap` is not set.
             let a_check =
                 a_swapped - b.clone() * swap.clone() - a.clone() * (one.clone() - swap.clone());
 
             // b_swapped - a ⋅ swap - b ⋅ (1-swap) = 0
             // This checks that `b_swapped` is equal to `a` when `swap` is set,
             // but remains as `b` when `swap` is not set.
-            let b_check = b_swapped - a * swap.clone() - b * (one.clone() - swap.clone());
+            let b_check = b_swapped - a * swap.clone() - b * (one - swap.clone());
```

Comment:
Aside: Here's an interesting constraint where a ternary helper that ensures its halves are either positive or zero could be used (strengthening the argument for the ternary helper being built around `1 - x` instead of `x - 1`).